marlowe-playground-client/src/Simulation.purs

```diff
@@ -253,6 +278,79 @@ handleAction settings AnalyseReachabilityContract = do
 
 handleAction _ Save = pure unit
 
+setOraclePrice ::
+  forall m.
+  MonadAff m =>
+  SPSettings_ SPParams_ -> HalogenM State Action ChildSlots Void m Unit
+setOraclePrice settings = do
+  exState <- use (_currentMarloweState <<< _executionState)
+  case exState of
+    Just es -> do
+      let
+        (Parties actions) = es.possibleActions
+      case Map.lookup (Role "kraken") actions of
+        Just acts -> do
+          case Array.head (Map.toUnfoldable acts) of
+            Just (Tuple (ChoiceInputId choiceId@(ChoiceId pair _) bounds) _) -> do
+              price <- getPrice settings "kraken" pair
+              liftEffect $ log ("setting choice " <> show choiceId <> " with price " <> show price)
+              handleAction settings (SetChoice choiceId price)
+            _ -> pure unit
+        Nothing -> pure unit
+    Nothing -> pure unit
+
+getPrice :: forall m. MonadAff m => SPSettings_ SPParams_ -> String -> String -> HalogenM State Action ChildSlots Void m BigInteger
+getPrice settings exchange pair = do
+  result <- runAjax (getPriceAjax settings exchange pair)
+  a <-
+    liftEffect case result of
+      NotAsked -> do
+        log "NotAsked"
+        pure 0.0
+      Loading -> do
+        log "Loading"
+        pure 0.0
+      Failure e -> do
+        log $ "Failure" <> errorToString e
+        pure 0.0
+      Success a -> pure a.price
+  let
+    price = fromMaybe (fromInt 0) (fromString (show (floor a)))
```

Comment:
1. I think it's nicer to use `zero` rather than `fromInt 0`
2. oh this Number -> String -> Number thing is nasty